src/test/rgw/test_rgw_iam_policy.cc

```diff
@@ -0,0 +1,504 @@
+		  s3GetObjectVersion | s3GetObjectAcl | s3GetObjectVersionAcl |
+		  s3GetObjectTorrent | s3GetObjectVersionTorrent |
+		  s3GetAccelerateConfiguration | s3GetBucketAcl |
+		  s3GetBucketCORS | s3GetBucketVersioning |
+		  s3GetBucketRequestPayment | s3GetBucketLocation |
+		  s3GetBucketPolicy | s3GetBucketNotification |
+		  s3GetBucketLogging | s3GetBucketTagging |
+		  s3GetBucketWebsite | s3GetLifecycleConfiguration |
+		  s3GetReplicationConfiguration);
+
+  EXPECT_EQ(p.eval(em, none, s3PutBucketPolicy,
+		   ARN(Partition::aws, Service::s3,
+		       "", arbitrary_tenant, "mybucket")),
+	    Effect::Allow);
+
+  EXPECT_EQ(p.eval(em, none, s3PutBucketPolicy,
+		   ARN(Partition::aws, Service::s3,
+		       "", arbitrary_tenant, "mybucket")),
+	    Effect::Allow);
+
+
+  for (auto i = 0ULL; i < s3Count; ++i) {
+    auto op = 1ULL << i;
+    if ((op == s3ListAllMyBuckets) || (op == s3PutBucketPolicy)) {
+      continue;
+    }
+
+    EXPECT_EQ(p.eval(em, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "confidential-data")),
+	      Effect::Pass);
+    EXPECT_EQ(p.eval(tr, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "confidential-data")),
+	      op & s3allow ? Effect::Allow : Effect::Pass);
+    EXPECT_EQ(p.eval(fa, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "confidential-data")),
+	      Effect::Pass);
+
+    EXPECT_EQ(p.eval(em, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "confidential-data/moo")),
+	      Effect::Pass);
+    EXPECT_EQ(p.eval(tr, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "confidential-data/moo")),
+	      op & s3allow ? Effect::Allow : Effect::Pass);
+    EXPECT_EQ(p.eval(fa, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "confidential-data/moo")),
+	      Effect::Pass);
+
+    EXPECT_EQ(p.eval(em, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "really-confidential-data")),
+	      Effect::Pass);
+    EXPECT_EQ(p.eval(tr, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "really-confidential-data")),
+	      Effect::Pass);
+    EXPECT_EQ(p.eval(fa, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant, "really-confidential-data")),
+	      Effect::Pass);
+
+    EXPECT_EQ(p.eval(em, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant,
+			 "really-confidential-data/moo")), Effect::Pass);
+    EXPECT_EQ(p.eval(tr, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant,
+			 "really-confidential-data/moo")), Effect::Pass);
+    EXPECT_EQ(p.eval(fa, none, op,
+		     ARN(Partition::aws, Service::s3,
+			 "", arbitrary_tenant,
+			 "really-confidential-data/moo")), Effect::Pass);
+
+  }
+}
+
+const string PolicyTest::arbitrary_tenant = "arbitrary_tenant";
+string PolicyTest::example1 = " \
```

Comment:
Hmm, maybe we could use here the raw string literals of C++11 similarly like we did in [`RGWSwiftWebsiteListingFormatter`](https://github.com/ceph/ceph/blob/master/src/rgw/rgw_formats.cc#L308). On the other hand, editors still experiences issues with highlighting such code.